api/envoy/api/v2/route/route.proto

```diff
@@ -298,6 +298,11 @@ message RouteMatch {
   // query parameters is nonzero, they all must match the *path* header's
   // query string for a match to occur.
   repeated QueryParameterMatcher query_parameters = 7;
+
+  // Specifies that only gRPC requests should be matched. The router will check
+  // that the content-type header has a application/grpc or one of the various
+  // application/grpc+ values.
+  bool is_grpc = 8;
```

Comment:
instead of a `bool` here, can we use a future extensible message? Some possible extension would be StringMatch on proto service name and/or method name, or message-type etc. We can start with google.protobuf.Empty first and then change to custom defined message.

Reply:
Yeah that's a good idea, I'll update

Reply:
FWIW this is a bit odd to me as a reader. What about just putting the bool inside of a oneof for clarity that we might want to add other types of matching in the future?

Reply:
Sorry or do we mean that we want to allow for different types of gRPC matching? If the latter maybe just create an empty grpc matcher object for clarity?

Reply:
I already pushed a commit that changes this into using an a `protobuf.Empty`, is that different from what you're talking about? Would you prefer a new, empty message instead?

Reply:
I was just suggesting that `Empty` is a little confusing to me and maybe an empty actual message called something like `GrpcRouteMatchOptions` or something like that might be more readable? It's not a big deal, just throwing it out there.